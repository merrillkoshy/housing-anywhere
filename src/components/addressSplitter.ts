const addressSplitter = (address:string) => { 
    type Searchable = {
        [key: string]: string;
    };
    
        const chunk = address
            .replace("(", "")
            .replace(")", "")
            .replace('"', "")
            .split(",");
    
        const proto: Searchable = {
            housenumber: "",
            street: "",
            city: "",
            state: "",
            postalCode: "",
            neighborhood: "",
            countryCode: "",
        };
        chunk.forEach((element, i) => {
            var currentKey = Object.keys(proto)[i];
            proto[currentKey] = element
                .replace('"', "")
                .trim()
                .replace('"', "");
        } );
    return proto;
}
export default addressSplitter;