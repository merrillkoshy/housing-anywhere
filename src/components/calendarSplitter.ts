const calendarSplitter = (cal: string) => {
	type Searchable = {
		[key: string]: string | number | Array<Searchable>;
	};

	const regex = new RegExp(/\(([^()]*)\)/g);
	const isAnArray = regex.test(cal);
	const parentArrayPusher = [];
	const childArrayPusher: Array<any> = [];

	const chunk = cal
		.replace("(", "")
		.replace(")", "")
		.replace('"', "")
		.split(",");

	const proto: Searchable = {
		type: 0,
		dateFrom: "",
		dateTo: "",
	};
	chunk.forEach((element, i) => {
		var currentKey = Object.keys(proto)[i];
		const isAnArray = regex.test(element);
		proto[currentKey] = element
			.replace('"', "")
			.trim()
			.replace('"', "")
			.replace("{", "")
			.replace("}", "");

		if (isAnArray) {
			childArrayPusher.push(proto[currentKey]);
			proto[currentKey] = childArrayPusher;
		}
	});
	if (isAnArray) {
		parentArrayPusher.push(proto);
		return parentArrayPusher;
	} else {
		return proto;
	}
};
export default calendarSplitter;
