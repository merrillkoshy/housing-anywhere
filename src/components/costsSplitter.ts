const costsSplitter = (address:string) => { 
    type Searchable = {
        [key: string]: string|number|boolean,
    };
    
        const chunk = address
            .replace( "(", "" )
            .trim()
            .replace( ")", "" )
            .trim()
            .replace( '"', "" )
            .trim()
            .replace( '\\"', "" )
            .replaceAll(/\s/g,'')
            .split(",");
    
        const proto: Searchable = {
            type: "",
            value: 0,
            payableAt: "",
            payableBy: "",
            refundable: "",
            estimated: "",
            mandatory: ""
        };
        chunk.forEach((element, i) => {
            var currentKey = Object.keys(proto)[i];
            proto[ currentKey ] = element
            .replace( "(", "" )
            .trim()
                .replace( ")", "" )
                .replace( "{", "" )
                .replace( "}", "" )
                .replace( '"', "" )
                .replace( ' ', "" )
                .trim()
                .replace( '"', "" )
                .replaceAll( /\s/g, '' );
                if ( proto[ currentKey ] === "t" ) proto[ currentKey ] = true;
                if ( proto[ currentKey ] === "f" ) proto[ currentKey ] = false;

        } );
    return proto;
}
export default costsSplitter;