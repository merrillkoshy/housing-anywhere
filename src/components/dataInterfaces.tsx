import React from "react";

interface Address {
	housenumber: string;
	street: string;
	city: string;
	state: string;
	postalCode: string;
	neighborhood: string;
	countryCode: string;
}
interface Rules {
	pets: string;
	cleaningCommonRoomsIncluded: boolean;
	cleaningPrivateRoomIncluded: boolean;
	playMusic: string;
	smoking: string;
	couple: boolean;
}

interface AdditionalCosts {
	water: boolean;
	gas: boolean;
	electricity: boolean;
	internet: boolean;
}
interface Facilities {
	allergyFriendly: boolean;
	housematesGender: string;
	registrationPossible: boolean;
	tenantStatus: string;
	freePlaces: number;
	bedrooms: number;
	bedroomFurnished: boolean;
	bedroomSize: number;
	bedroomLock: boolean;
	balconyTerrace: string;
	basement: string;
	bathroom: string;
	garden: string;
	kitchen: string;
	livingRoom: string;
	parking: string;
	toilet: string;
	totalSize: number;
	wheelchairAccessible: boolean;
	airConditioning: boolean;
	bed: boolean;
	closet: boolean;
	desk: boolean;
	dishwasher: boolean;
	dryer: boolean;
	internet: boolean;
	roomFurniture: boolean;
	tv: boolean;
	washingMachine: boolean;
	wifi: boolean;
	flooring: string;
	heating: string;
	kitchenware: string;
	additionalCosts: AdditionalCosts;
}

interface Costs {
	type: string;
	value: string;
	payableAt: string;
	payableBy: string;
	refundable: boolean;
	estimated: boolean;
	mandatory: boolean;
}
interface Images {
	url: string;
}
interface CalendarOperations {
	type: number;
	dateFrom: string;
	dateTo: string;
}
export interface HAData {
	price: number;
	currencyCode: string;
	address: Address;
	type: number;
	kind: number;
	description: string;
	deposit: number;
	estimatedBills: number;
	minimumStayMonths: string;
	maxBookableDays: number;
	moveInWindow: number;
	currentOccupancy: number;
	rules: Rules;
	minAge: number;
	maxAge: number;
	preferredGender: string;
	alias: string;
	externalReference: string;
	extraData: string;
	facilities: Facilities;
	calendarOperations: CalendarOperations[];
	images: Images[];
	costs: Costs[];
}
