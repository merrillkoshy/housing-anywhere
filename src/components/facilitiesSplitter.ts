const facilitiesSplitter = (facilities: string) => {
	const proto: Searchable = {
		allergyFriendly: false,
		housematesGender: "",
		registrationPossible: false,
		tenantStatus: "",
		freePlaces: 0,
		bedrooms: 0,
		bedroomFurnished: false,
		bedroomSize: 0,
		bedroomLock: false,
		balconyTerrace: "",
		basement: "",
		bathroom: "",
		garden: "",
		kitchen: "",
		livingRoom: "",
		parking: "",
		toilet: "",
		totalSize: 0,
		wheelchairAccessible: false,
		airConditioning: false,
		bed: false,
		closet: false,
		desk: false,
		dishwasher: false,
		dryer: false,
		internet: false,
		roomFurniture: false,
		tv: false,
		washingMachine: false,
		wifi: false,
		flooring: "",
		heating: "",
		kitchenware: "",
	};
	const child: Searchable = {
		water: false,
		gas: false,
		electricity: false,
		internet: false,
	};

	if (facilities === null || facilities === "") return { proto, child };
	type Searchable = {
		[key: string]: string | boolean | number;
	};

	const chunk = facilities
		.replace("(", "")
		.trim()
		.replace(")", "")
		.trim()
		.replace('"', "")
		.trim()
		.replace('\\"', "")
		.replaceAll(/\s/g, "")
		.split(",");
	const additionalCosts = facilities
		.split('"(')[1]
		.replace(')"', "")
		.replace(")", "")
		.split(",");

	chunk.forEach((element, i) => {
		var currentKey = Object.keys(proto)[i];
		proto[currentKey] = element
			.replace("(", "")
			.trim()
			.replace(")", "")
			.replace('"', "")
			.replace(" ", "")
			.trim()
			.replace('"', "")
			.replaceAll(/\s/g, "");
		if (proto[currentKey] === "t") proto[currentKey] = true;
		if (proto[currentKey] === "f") proto[currentKey] = false;
	});
	additionalCosts.forEach((element, i) => {
		var currentKey = Object.keys(proto)[i];
		child[currentKey] = element.trim();
		if (child[currentKey] === "t") proto[currentKey] = true;
		if (child[currentKey] === "f") proto[currentKey] = false;
	});
	return { proto, child };
};
export default facilitiesSplitter;
