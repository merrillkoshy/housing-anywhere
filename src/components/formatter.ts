const formatter = (element: any): string => {
	var temp = "";
	Object.values(element).map((val: any, i: number) => {
		if (i !== Object.keys(element).length - 1) {
			if (val === "") return (temp = temp + "null,");
			return (temp = temp + val.toString() + ",");
		}
		if (typeof val === "object")
			return (temp = temp + '"' + formatter(val) + '"');
		if (val === "") return (temp = temp + "null");
		return (temp = temp + val);
	});
	return "(" + temp + ")";
};
const arrayformatter = (element: any): any[] => {
	var temp = "";
	var innertemp = [];
	innertemp.push(
		element.map((el: any) => {
			Object.values(el).map((val: any, i: number) => {
				if (i !== Object.keys(el).length - 1)
					return (temp = temp + val.toString() + ",");
				return (temp = temp + val);
			});
			return "(" + temp + ")";
		})
	);

	return innertemp.flat();
};
const formatters = { formatter, arrayformatter };
export default formatters;
