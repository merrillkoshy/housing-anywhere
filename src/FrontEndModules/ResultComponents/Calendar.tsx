import React from "react";
import { CalendarOperations } from "../../components/dataInterfaces";
import { v4 as uuidv4 } from "uuid";

const Calendar = ({ calendar }: { calendar: Array<CalendarOperations> }) => {
	return (
		<div className="d-flex flex-column my-2">
			{calendar &&
				calendar?.map((cals) => {
					return (
						<div key={uuidv4()}>
							<div className="row">
								<div className="col">
									{cals?.type ? (
										<p id="typeCal" className="text-danger">
											Unavailable
										</p>
									) : (
										<p id="typeCal" className="text-success">
											Available
										</p>
									)}
								</div>
							</div>
							<div className="row">
								<div className="col">
									<strong id="dateFrom">From:</strong>
									{cals?.dateFrom}
								</div>
								<div className="col">
									<strong id="dateTo">To:</strong>
									{cals?.dateTo}
								</div>
							</div>
						</div>
					);
				})}
		</div>
	);
};
export default Calendar;
