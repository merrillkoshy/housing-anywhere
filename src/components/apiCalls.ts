import axios from "axios";
import { toast } from "react-toastify";
import { HAData } from "./dataInterfaces";
import formatter from "./formatter";

const insertData = (data: HAData, getter: () => void) => {
	var {
		id,
		price,
		currencycode,
		address,
		type,
		kind,
		description,
		deposit,
		estimatedbills,
		minimumstaymonths,
		maxbookabledays,
		moveinwindow,
		currentoccupancy,
		rules,
		minage,
		maxage,
		preferredgender,
		alias,
		externalreference,
		extradata,
		facilities,
		calendaroperations,
		images,
		costs,
	} = data;

	//since we dont have an image server, we do temp images here
	images = [{ url: "https://picsum.photos/300/300" }];

	//formatting for psql because js arrays and objects are notated
	//differently for psql eg: for arrays its ARRAY[(val1,val2)] and
	//for objects {} will become ().
	const stringedAddress = formatter.formatter(address);

	const stringedFacilities = formatter.formatter(facilities);

	const stringedRules = formatter.formatter(rules);

	const stringedImages = formatter.arrayformatter(images);

	const stringedCosts = formatter.arrayformatter(costs);

	const stringedcalendaroperations = formatter.arrayformatter(
		calendaroperations
	);

	const toSend = {
		id,
		price,
		currencycode,
		address: stringedAddress,
		type,
		kind,
		description,
		deposit,
		estimatedbills,
		minimumstaymonths,
		maxbookabledays,
		moveinwindow,
		currentoccupancy,
		rules: stringedRules,
		minage,
		maxage,
		preferredgender,
		alias,
		externalreference,
		extradata,
		facilities: stringedFacilities,
		calendaroperations: stringedcalendaroperations,
		images: stringedImages,
		costs: stringedCosts,
	};

	axios
		.post(`https://ha-server.herokuapp.com/listing`, toSend, {
			headers: {
				"Access-Control-Allow-Headers": "Content-Type, jwtSecret",
				jwtSecret: "merrillkoshy",
				"Access-Control-Allow-Origin": "*",
				"Content-Type": "application/json",
			},
		})
		.then((res) => {
			toast(`Inserted listing with id ${res.data.id}`, {
				position: "top-center",
				type: "success",
			});
			getter();
		})
		.catch((err) => {
			console.log(err);
			toast(`Error in call`, {
				position: "top-center",
				type: "error",
			});
		});
};

const updateListing = async (id: number, field: any) => {
	const { price, currencyCode } = field;
	await axios
		.put(
			`https://ha-server.herokuapp.com/listing`,
			{
				price,
				currencyCode,
				id,
			},
			{
				headers: {
					"Access-Control-Allow-Headers": "Content-Type, jwtSecret",
					jwtSecret: "merrillkoshy",
					"Access-Control-Allow-Origin": "*",
					"Content-Type": "application/json",
				},
			}
		)
		.then((res) => {
			toast("Succesfully Updated ", {
				position: "bottom-center",
				type: "success",
			});
		})
		.catch((err) => {
			console.log(err);
			toast(`Error in call`, {
				position: "bottom-center",
				type: "error",
			});
		});
};
const deleteListing = async (id: number, getter: () => void) => {
	await axios
		.delete(`https://ha-server.herokuapp.com/listing/${id}`, {
			headers: {
				"Access-Control-Allow-Headers": "Content-Type, jwtSecret",
				jwtSecret: "merrillkoshy",
				"Access-Control-Allow-Origin": "*",
				"Content-Type": "application/json",
			},
		})
		.then((res) => {
			toast("Succesfully Deleted ", {
				position: "bottom-center",
				type: "success",
			});
			getter();
		})
		.catch((err) => {
			console.log(err);
			toast(`Error in call`, {
				position: "bottom-center",
				type: "error",
			});
		});
};
const apiCalls = { insertData, deleteListing, updateListing };

export default apiCalls;
