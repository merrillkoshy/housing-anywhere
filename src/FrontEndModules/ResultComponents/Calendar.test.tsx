import React from "react";
import { shallow } from "enzyme";
import Calendar from "./Calendar";
import { CalendarOperations } from "../../components/dataInterfaces";

describe("CostsModule module", () => {
	const calendarValue: Array<CalendarOperations> = [
		{
			type: 0,
			dateFrom: "",
			dateTo: "",
		},
	];
	const CalendarOperationsModule = shallow(
		<Calendar calendar={calendarValue} />
	);
	it("renders", () => {
		expect(CalendarOperationsModule).toMatchSnapshot();
	});

	it("finds an li typeCal", () => {
		expect(CalendarOperationsModule.find("#typeCal").exists()).toBe(true);
	});
	it("finds an li dateFrom", () => {
		expect(CalendarOperationsModule.find("#dateFrom").exists()).toBe(true);
	});
	it("finds an li dateTo", () => {
		expect(CalendarOperationsModule.find("#dateTo").exists()).toBe(true);
	});
});
