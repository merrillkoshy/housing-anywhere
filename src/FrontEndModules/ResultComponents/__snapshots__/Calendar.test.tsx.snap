// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CostsModule module renders 1`] = `
<div
  className="d-flex flex-column my-2"
>
  <div
    key="82e6ce21-8b9e-40bf-827b-872bb552f589"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <p
          className="text-success"
          id="typeCal"
        >
          Available
        </p>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <strong
          id="dateFrom"
        >
          From:
        </strong>
      </div>
      <div
        className="col"
      >
        <strong
          id="dateTo"
        >
          To:
        </strong>
      </div>
    </div>
  </div>
</div>
`;
